components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/RegistrationFlowEngine.java

```diff
@@ -0,0 +1,173 @@
+import org.wso2.carbon.identity.user.registration.engine.graph.UserChoiceDecisionNode;
+import org.wso2.carbon.identity.user.registration.mgt.Constants;
+import org.wso2.carbon.identity.user.registration.mgt.exception.RegistrationFrameworkException;
+import org.wso2.carbon.identity.user.registration.mgt.model.DataDTO;
+import org.wso2.carbon.identity.user.registration.mgt.model.NodeConfig;
+import org.wso2.carbon.identity.user.registration.engine.model.Response;
+import org.wso2.carbon.identity.user.registration.engine.model.RegistrationContext;
+import org.wso2.carbon.identity.user.registration.mgt.model.RegistrationGraphConfig;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ErrorMessages.ERROR_CODE_FIRST_NODE_NODE_FOUND;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ErrorMessages.ERROR_CODE_UNSUPPORTED_NODE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_COMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_INCOMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_PROMPT_ONLY;
+import static org.wso2.carbon.identity.user.registration.engine.util.RegistrationFlowEngineUtils.handleServerException;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.StepTypes.REDIRECTION;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.StepTypes.VIEW;
+
+/**
+ * Engine to execute the registration flow.
+ */
+public class RegistrationFlowEngine {
+
+    private static final Log LOG = LogFactory.getLog(RegistrationFlowEngine.class);
+
+    private static final RegistrationFlowEngine instance = new RegistrationFlowEngine();
+
+    private RegistrationFlowEngine() {
+
+    }
+
+    public static RegistrationFlowEngine getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Execute the registration sequence.
+     *
+     * @param context Registration context.
+     * @return Node response.
+     * @throws RegistrationFrameworkException If an error occurs while executing the registration sequence.
+     */
+    public RegistrationStep execute(RegistrationContext context)
+            throws RegistrationFrameworkException {
+
+        RegistrationGraphConfig graph = context.getRegGraph();
+
+        String tenantDomain = context.getTenantDomain();
+        if (graph.getFirstNodeId() == null) {
+            throw handleServerException(ERROR_CODE_FIRST_NODE_NODE_FOUND, graph.getId(), tenantDomain);
+        }
+
+        NodeConfig currentNode = context.getCurrentNode();
+        if (currentNode == null) {
+            LOG.debug("Current node is not set. Setting the first node as the current node and starting the " +
+                              "registration sequence.");
+            currentNode = graph.getNodeConfigs().get(graph.getFirstNodeId());
```

Comment:
Where do we set the current node to context

Reply:
It's set within the while loop. (Line 92 and 101)